### PR TITLE
Add mdns browser

### PR DESCRIPTION
**Pull request by FranziskaKnoedtel**

### CODE REVIEW

File: core/mdns_browser.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+			return s, err
+		}
+	}
+}
+
+// Start mdns server and browse interfaces for mdns services with the specified service name.
+// Sort out the service with the specified instance name.
+// Return the ip address and port of the found service
+// TODO get additional ports from txt field
+func GetAddressFromService(instanceName string, serviceName string, timeout int) (string, uint16, error) {
```
Comment (KlausPopp) [inline core/mdns_browser.go:37]: Install pre-commit-hooks. 
If they are installed, they would warn you that the function comment must begin with the function name:
// GetAddressFromService queries mdns for the requested....
What exactly has to be passed to instanceName and serviceName? Can I specify the protocol (tcp/udp)?

File: core/mdns_browser.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+	conn, err := dbus.SystemBus()
+	if err != nil {
+		log.Fatalf("Cannot get system bus: %v", err)
```
Comment (KlausPopp) [inline core/mdns_browser.go:40]: You are in a package. Never exit from a function in a package. Leave this up to the package user whether to treat this as a fatal error!

File: core/mdns_browser.go [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+	}
+
+	server, err := avahi.ServerNew(conn)
+	if err != nil {
+		log.Fatalf("Avahi new failed: %v", err)
```
Comment (KlausPopp) [inline core/mdns_browser.go:45]: You are in a package. Never exit from a function in a package. Leave this up to the package user whether to treat this as a fatal error!

File: core/mdns_browser.go [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+	}
+
+	sb, err := server.ServiceBrowserNew(avahi.InterfaceUnspec, avahi.ProtoUnspec, serviceName, "local", 0)
+	if err != nil {
+		log.Fatalf("ServiceBrowserNew() failed: %v", err)
```
Comment (KlausPopp) [inline core/mdns_browser.go:50]: You are in a package. Never exit from a function in a package. Leave this up to the package user whether to treat this as a fatal error!

File: core/mdns_browser.go [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+// Search the for the service with the specified instance name
+func searchService(sb *avahi.ServiceBrowser, srv *avahi.Server, name string, t int) (avahi.Service, error) {
+	var s avahi.Service
+	for {
+		select {
+		case s = <-sb.AddChannel:
+			s, err := srv.ResolveService(s.Interface, s.Protocol, s.Name,
+				s.Type, s.Domain, avahi.ProtoUnspec, 0)
+			if err != nil {
+				log.Fatalf("ResolveService() failed: %v", err)
```
Comment (KlausPopp) [inline core/mdns_browser.go:21]: You are in a package. Never exit from a function in a package. Leave this up to the package user whether to treat this as a fatal error!

**Review by KlausPopp — CHANGES_REQUESTED** [review]

**Comment by KlausPopp** [comment]
Please add in core: NewClientFromService() function (That calls the previous NewClientFromSocketAddress).
Update Examples in README

File: README.md [hunk 6]
```diff
@@ -95,6 +95,44 @@ func main() {
 }
 ```
 
+### Connect to a Target with service name
```
Comment (batthebee) [inline README.md:98]: Add system dependencies to readme, like dbus.

**Review by KlausPopp — APPROVED** [review]

**Pull request merged**